### PR TITLE
Cleanup websocket and add benchmark

### PR DESCRIPTION
**Pull request by ppamorim**
This PR simplifies the websocket deserializing and adds a benchmark test against the websocket deserializer.

### CODE REVIEW

File: src/websockets.rs [hunk 1]
```diff
@@ -109,57 +118,38 @@ impl<'a> WebSockets<'a> {
     pub fn disconnect(&mut self) -> Result<()> {
         if let Some(ref mut socket) = self.socket {
             socket.0.close(None)?;
-            Ok(())
-        } else {
-            bail!("Not able to close the connection");
+            return Ok(());
         }
+        bail!("Not able to close the connection");
```
Comment (dorak88783) [inline src/websockets.rs:123]: I don't get this part of the change, it looks like the new code will always hit this `bail!` statement.

File: benches/websocket_benchmark.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    });
+
+    let mut web_socket: WebSockets<'_> = WebSockets::new(|event: WebsocketEvent| {
+        Ok(())
+    });
+
+    group.sample_size(200);
+    group.measurement_time(Duration::new(35, 0));
+    group.bench_function("handle_msg all symbols", |b| b.iter(|| handle_msg(&mut web_socket_subscribed, &all_symbols_json)));
+    group.bench_function("handle_msg BTCUSDT symbol", |b| b.iter(|| handle_msg(&mut web_socket, &btc_symbol_json)));
```
Comment (dorak88783) [inline benches/websocket_benchmark.rs:34]: Can't you write directly `web_socket.test_handle_message(&btc_symbol_json)` and hence the `handle_msg` function is not needed?

File: src/websockets.rs (hunk 1 above)
Comment (ppamorim) [reply src/websockets.rs:123]: No it won't because the `return Ok(());` causes  a early return. 

https://medium.com/swlh/return-early-pattern-3d18a41bba8

Comment (dorak88783) [reply src/websockets.rs:123]: Oops, missed that line. Thanks!

**Pull request merged**